lstchain/hdu/hdu_table.py

```
@@ -0,0 +1,350 @@
+def create_obs_hdu_index(filename_list, fits_dir):
+    """
+    Create the obs table and hdu table (below some explanation)
+
+    A two-level index file scheme is used in the IACT community to allow an arbitrary folder structures
+    For each directory tree, two files should be present:
+    **obs-index.fits.gz**
+    (defined in http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/obs_index/index.html)
+    **hdu-index.fits.gz**
+    (defined in http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/hdu_index/index.html)
+
+    obs-index contains the single run informations e.g.
+    (OBS_ID, RA_PNT, DEC_PNT, ZEN_PNT, ALT_PNT)
+    while hdu-index contains the informations about the locations of the other HDU (header data units)
+    necessary to the analysis e.g. A_eff, E_disp and so on...
+        http://gamma-astro-data-formats.readthedocs.io/en/latest/
+
+    This function will create the necessary data format, starting from the path that contains the DL3
+    converted fits file.
+
+    Parameters
+    ----------
+    filename_list : list
+        list of filenames
+    fits_dir : str
+        directory containing the fits file
+    """
+
+    # empty lists with the quantities we want
+    hdu_tables = []
+
+    obs_id = []
+    ra_pnt = []
+    dec_pnt = []
+    zen_pnt = []
+    alt_pnt = []
+    az_pnt = []
+    ontime = []
+    livetime = []
+    deadc = []
+    tstart = []
+    tstop = []
+    source = []
+    N_TELS = []
+    TELLIST = []
+
+    # loop through the files
+    for file in filename_list:
+
+        if os.path.exists(str(fits_dir)+"/"+file):
+            try:
+                event_table = Table.read(str(fits_dir)+"/"+file, hdu="EVENTS")
+                gti_table = Table.read(str(fits_dir)+"/"+file, hdu="GTI")
+                pointing_table = Table.read(str(fits_dir)+"/"+file, hdu="POINTING")
+            except Exception:
+                print(f"fits corrupted for file {file}")
```

Broken data files should not be ignored I think. This should raise an exception!

Also, use logging, not print statements.

---

Yes, I had forgotten about this. Will amend it.